src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -1034,10 +1034,10 @@ protected void GetFeatureIniContent(RoleMappedData data, ref VBuffer<ReadOnlyMem
         /// <param name="upperBounds">The bin upper bounds, maximum length will be <paramref name="maxBins"/></param>
         /// <returns>Whether finding the bins was successful or not. It will be unsuccessful iff <paramref name="values"/>
         /// has any missing values. In that event, the out parameters will be left as null.</returns>
-        protected static bool CalculateBins(BinFinder binFinder, ref VBuffer<double> values, int maxBins, int minDocsPerLeaf,
+        protected static bool CalculateBins(BinFinder binFinder, in VBuffer<double> values, int maxBins, int minDocsPerLeaf,
             ref double[] distinctValues, ref int[] distinctCounts, out double[] upperBounds)
```

Comment:
>ref double[] distinctValues, ref int[] distinctCounts, [](start = 12, length = 54)

Can you ask @TomFinley why we have them? I don't see any usages of them.

Reply:
I think this from a prior time before @costin-eseanu (I think?) introduced the concept of the `BinFinder`, which internally incorporates this concept of a reusable buffer. Yet somehow these arguments were not removed at the same time. Or maybe I made the mistake, since I also touched the disk implementation a lot, so maybe I did this. Anyway this was done many years ago.

Not quite relevant to the PR, but something we may want to remove opportunistically. It's not really hurting anything exactly, but it is a little silly to keep these around now that we do have bin finder.

Reply:
I will remove them. Might as well do some clean up while I'm here.

---
In reply to: [229529149](https://github.com/dotnet/machinelearning/pull/1454#discussion_r229529149) [](ancestors = 229529149)